### PR TITLE
FIX-#4376: Upgrade pandas to 1.4.2

### LINKED ISSUE #4376: Upgrade pandas to 1.4.2

### PR DESCRIPTION
**Pull request by Garra1980**
Signed-off-by: izamyati <igor.zamyatin@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4376 
- [ ] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-pandas==1.4.1
+pandas==1.4.2
```
Comment (dchigarev) [inline requirements-dev.txt:1]: There are also other files where we have to change the version in, you can check the previous PR for the full list of them #4235

Comment (Garra1980) [reply requirements-dev.txt:1]: will do, thanks!

**Comment by YarShev** [comment]
@Garra1980, please add a release note in `docs/release_notes/release_notes-0.15.0.rst`.

**Comment by Garra1980** [comment]
> @Garra1980, please add a release note in `docs/release_notes/release_notes-0.15.0.rst`.

DOne

**Review by dchigarev — APPROVED** [review]

**Pull request merged**